src/installer/corehost/cli/bundle/info.h

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __INFO_H_
+#define __INFO_H_
+
+#include "error_codes.h"
+#include "header.h"
+
+// bundle::info supports:
+// * API for identification of a single-file app bundle, and
+// * Minimal probing and mapping functionality only for the app.runtimeconfig.json and app.deps.json files.
+// bundle::info is used by HostFxr to read the above config files.
+
+namespace bundle
+{
+    struct info_t
+    {
+        struct config_t
+        {
+            config_t() {}
+
+            config_t(const config_t& config)
+            {
+                m_path = config.m_path;
+                m_location = config.m_location;
+            }
+
+            config_t(const pal::string_t& path, const location_t *location)
+            {
+                m_path = path;
+                m_location = location;
+            }
+
+            bool matches(const pal::string_t& path) const
+            {
+                return m_location->is_valid() && path.compare(m_path) == 0;
+            }
+
+            static bool probe(const pal::string_t& path)
+            {
+                return is_single_file_bundle() && 
+                       (the_app->m_deps_json.matches(path) || the_app->m_runtimeconfig_json.matches(path));
+            }
+
+            static const int8_t* map(const pal::string_t& path, const location_t* &location);
+            static void unmap(const int8_t* addr, const location_t* location);
+
+        private:
+            pal::string_t m_path;
+            const location_t *m_location;
+        } json_info;
```

Comment:
`json_info` doesn't seem to be used from a quick glance.  Might want to remove the `config_t()` constructor, then, because method such as `matches()` will reference a garbage pointer